### PR TITLE
Eliminate warnings in downstream projects

### PR DESCRIPTION
**Pull request by jensmithorcio**
- Add shouldForwardProp config to select component styled divs to remove warning re invalid div attributes (width, height etc)
- Use styled-system props to ensure proptypes validation passes for valid styled-system properties

### CODE REVIEW

File: lib/components/Select/index.js [hunk 1]
```diff
@@ -6,29 +6,34 @@ import { default as ReactCreateableSelect } from "react-select/creatable";
...
+const Wrapper = styled("div")
+  .withConfig({ shouldForwardProp })
+  .attrs((props) => ({
+    "data-testid": props["data-testid"] ? props["data-testid"] : null
+  }))(
+    css({
+      display: "inline-block",
+      width: "100%"
+    }),
+    SelectStyles
```
Comment (KaranAtTeamForm) [inline lib/components/Select/index.js:23]: Lint errors.

**Review by KaranAtTeamForm — APPROVED** [review]
LGTM :)

**Review by KaranAtTeamForm — APPROVED** [review]
Build failing because of audit-ci vulnerabilities.

**Review by KaranAtTeamForm — CHANGES_REQUESTED** [review]
Build failing because of audit-ci vulnerabilities.

**Review by KaranAtTeamForm — APPROVED** [review]

**Review by shomukai — APPROVED** [review]

**Review by deividasm — APPROVED** [review]

**Pull request merged**